dcpy/utils/s3.py

```diff
@@ -2,9 +2,15 @@
 from io import BytesIO
 from pathlib import Path
 import typer
-from typing import Any, Optional
+from typing import Any, Literal, TYPE_CHECKING
 import boto3
+from botocore.response import StreamingBody
 from botocore.client import Config
+
+if TYPE_CHECKING:
+    from mypy_boto3_s3.client import S3Client
+else:
+    S3Client = object
```

Comment:
This was sort of interesting. Makes sense though - for s3 typing we need the additional `boto-stubs` package with the `s3` option. Nice to not have to include an extra package in our build environments, only dev.

Reply:
are there any other typeing-related imports we'd also wanna put this condition around?

Reply:
Not that I can see